src/esn.calendar.libs/app/components/event/form/event-form.controller.js

```diff
@@ -588,4 +594,62 @@ function CalEventFormController(
       return calendar.getUniqueId() === uniqueId;
     });
   }
+
+  function duplicateEvent() {
+    // Build an event copy based on the currently edited event.
+    const duplicate = _generateEditedEventCopy();
+
+    // Set the alarm if there was any
+    // alarm can't be set with CalendarShell.fromIncompleteShell because the event needs to be created or cloned first.
+    if ($scope.editedEvent.alarm) {
+      duplicate.alarm = $scope.editedEvent.alarm;
+    }
+
+    // Reset the partstat from the original event.
+    _resetAttendeesParticipation(duplicate);
+
+    // Check and set a new video conference link if needed.
+    if (duplicate && duplicate.xOpenpaasVideoconference) {
+      // Wait for the VideoConfConfigurationService to generate a link.
+      return _generateVideoConferenceUrl().then(url => {
+        duplicate.xOpenpaasVideoconference = url;
+        _showDuplicateEventForm(duplicate);
+      });
+    }
+
+    _showDuplicateEventForm(duplicate);
+  }
+
+  function _showDuplicateEventForm(event) {
+    if (!event) return;
+    // Close the currently opened event form.
+    $scope.cancel();
+    // Open the duplicate event creation form after a short delay ( let the first modal finish hiding ).
+    $timeout(function() {
+      calOpenEventForm($scope.event.calendarHomeId, event);
+    }, CAL_EVENT_FORM_SPINNER_TIMEOUT_DURATION);
+  }
+
+  function _generateEditedEventCopy() {
+    const details = CAL_EVENT_DUPLICATE_KEYS
+      .reduce((details, key) => ($scope.editedEvent[key] ? { [key]: $scope.editedEvent[key], ...details } : details), {});
+
+    return CalendarShell.fromIncompleteShell(details);
+  }
+
+  function _generateVideoConferenceUrl() {
+
+    return VideoConfConfigurationService.getOpenPaasVideoconferenceAppUrl()
+      .then(openPaasVideoconferenceAppUrl => `${openPaasVideoconferenceAppUrl}${uuid4.generate()}`)
+      .catch(err => {
+        $log.error('Cannot generate a new video conference URL', err);
+        _displayNotification(notificationFactory.weakError, null, 'Failed to create a new video conference room');
```

Comment:
I guess this already has a translation, right?

Reply:
nope, added now